### PR TITLE
Update Draggable API Docs

### LINKED ISSUE #69479: Adopt a Widget - Draggable
The [Draggable](https://master-api.flutter.dev/flutter/widgets/Draggable-class.html) widget needs:

- A code / DartPad sample
For details about how to participate, see [flutter.dev/adoptawidget](https://flutter.dev/adoptawidget)

### PR DESCRIPTION
**Pull request by parzuko**
Adopt A Widget: Draggable

Added a code / DartPad sample to the Draggable API Docs

This pull request is:

- [x] A code sample
- [ ] More references
- [ ] More explanation

closes #69479

### CODE REVIEW

**Review by shihaohong — CHANGES_REQUESTED** [review]
Thanks for the contribution. It seems like there are some changes unrelated to the API doc due to the use of some Dart code formatter. Could you please revert those changes and keep only the API doc part of the change? Thanks!

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 1]
```diff
@@ -24,7 +24,8 @@ typedef DragTargetAccept<T> = void Function(T data);
 /// Signature for determining information about the acceptance by a [DragTarget].
 ///
 /// Used by [DragTarget.onAcceptWithDetails].
-typedef DragTargetAcceptWithDetails<T> = void Function(DragTargetDetails<T> details);
+typedef DragTargetAcceptWithDetails<T> = void Function(
+    DragTargetDetails<T> details);
```
Comment (yash1200) [inline packages/flutter/lib/src/widgets/drag_target.dart:28]: Revert these unrelated changes here, and others also

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 2]
```diff
@@ -98,6 +98,64 @@ enum DragAnchor {
 /// widget always displays [child].
 ///
 /// {@youtube 560 315 https://www.youtube.com/watch?v=QzA4c4QHZCY}
+/// 
+/// {@tool dartpad --template=stateless_widget_material}
```
Comment (shihaohong) [inline packages/flutter/lib/src/widgets/drag_target.dart:102]: This will need to be a stateful widget since you need to keep track of your `acceptedData`. Otherwise, it will keep resetting to 0 in your sample.
```suggestion
/// {@tool dartpad --template=stateful_widget_material}
```

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 3]
```diff
@@ -98,6 +98,64 @@ enum DragAnchor {
...
+///
+/// The follwing example has the [Draggable] widget along with the [DragTarget] in a row demonstrating what happens when you drag the element to the target.
```
Comment (shihaohong) [inline packages/flutter/lib/src/widgets/drag_target.dart:104]: Prefer lines < 80 characters. Also, fixed some typos and added a little clarification:
```suggestion
/// The following example has a [Draggable] widget along with a [DragTarget] 
/// in a row demonstrating an incremented `acceptedData` integer value when 
/// you drag the element to the target.
```

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 4]
```diff
@@ -98,6 +98,64 @@ enum DragAnchor {
...
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   var acceptedData = 0;
```
Comment (shihaohong) [inline packages/flutter/lib/src/widgets/drag_target.dart:108]: This will need to be moved into a State class outside of the build method.

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 5]
```diff
@@ -98,6 +98,64 @@ enum DragAnchor {
...
+///     children : [
+///       Draggable(
+///         // Data is the value this Draggable stores.
+///         data : 10,
+///         child : Container(
+///           height : 100.0,
+///           width : 100.0,
+///           color : Colors.lightGreenAccent,
+///           child : Center(
+///             child : Text("I am the Draggable")
```
Comment (shihaohong) [inline packages/flutter/lib/src/widgets/drag_target.dart:120]: There doesn't need to be a space before the colon. Applies to the rest of the docs:
```dart
data: 10,
```

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 6]
```diff
@@ -98,6 +98,64 @@ enum DragAnchor {
...
+///           )
```
Comment (shihaohong) [inline packages/flutter/lib/src/widgets/drag_target.dart:121]: Add trailing commas where needed, here and everywhere.
```suggestion
///           ),
```

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 7]
```diff
@@ -98,6 +98,64 @@ enum DragAnchor {
...
+///               width : 100.0,
+///               color : Colors.cyan,
+///               child : Center(
+///                 child : Text("Value is updated to: $acceptedData"),
+///               )
+///             )
+///          }
+///          onAccept : (int data){
+///            acceptedData += data;
+///          }
```
Comment (shihaohong) [inline packages/flutter/lib/src/widgets/drag_target.dart:151]: This will need to be wrapped in a `setState`. Otherwise, the UI will not update with the value changes.
```dart
            onAccept: (int data) {
              setState(() {
                acceptedData += data;
              });
            },
```

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 8]
```diff
@@ -802,4 +860,4 @@ class _DragAvatar<T extends Object> extends Drag {
     }
     return Offset(0.0, offset.dy);
   }
-}
+}
```
Comment (shihaohong) [inline packages/flutter/lib/src/widgets/drag_target.dart:863]: Files should end with a newline character
```suggestion
}

```

**Review by shihaohong — COMMENTED** [review]
I left some comments, mostly about styling and the use of a stateful widget.

Could we also discuss the idea of the sample? I understand that dragging the Draggable to the DragTarget results in incrementing the integer by 10. However, why is the child when dragging "I am what's left behind"?

**Comment by parzuko** [comment]
> I left some comments, mostly about styling and the use of a stateful widget.
> 
> Could we also discuss the idea of the sample? I understand that dragging the Draggable to the DragTarget results in incrementing the integer by 10. However, why is the child when dragging "I am what's left behind"?

Left Behind

This is what I had in mind. I thought "I am what's left behind " signifies the purpose of the childWhenDragging. Should I remove / edit it ?

**Comment by shihaohong** [comment]
@parzuko Ahh I think I get what you mean now. Maybe use a different phrase, like "This widget is shown when a drag is taking place." At first, I wasn't sure what "being left behind" meant. In reality, it's more so that a different widget is being displayed instead of the original one when a drag is taking place.

File: packages/flutter/lib/src/widgets/drag_target.dart (hunk 2 above)
Comment (shihaohong) [reply packages/flutter/lib/src/widgets/drag_target.dart:102]: For more info on how the templates look: https://github.com/flutter/flutter/blob/master/dev/snippets/config/templates/README.md#materialapp-templates

File: packages/flutter/lib/src/widgets/drag_target.dart (hunk 7 above)
Comment (parzuko) [reply packages/flutter/lib/src/widgets/drag_target.dart:151]: That's what I thought! However, regardless of it being wrapped in a stateless or statefull widget it doesn't seem to work.
Try this [DartPad ](https://dartpad.dev/b5c028995679069b7fd4b799f718bdef) for context.

**Comment by parzuko** [comment]
> @parzuko Ahh I think I get what you mean now. Maybe use a different phrase, like "This widget is shown instead when a drag is taking place." At first, I wasn't sure what "being left behind" meant. In reality, it's more so that a different widget is being displayed instead of the original one when a drag is taking place.

Hey @shihaohong, could you take a look at the DartPad example I mentioned for the setState part?

File: packages/flutter/lib/src/widgets/drag_target.dart (hunk 7 above)
Comment (shihaohong) [reply packages/flutter/lib/src/widgets/drag_target.dart:151]: It's because you're using a StatelessWidget here. If you update it into a StatefulWidget it works:
```dart

class Game extends StatefulWidget {
  @override
  _GameState createState() => _GameState();
}

class _GameState extends State<Game> {
  int acceptedData = 0;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Draggables"),
        backgroundColor: Colors.pink,
      ),
      body: Center(
        child: Row(
          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
          children: [
            Draggable<int>(
              data: 10,
              child: Container(
                height: 100.0,
                width: 100.0,
                color: Colors.pinkAccent,
                child: Center(
                  child: Text("Draggable"),
                ),
              ),
              feedback: Container(
                color: Colors.deepOrange,
                height: 100,
                width: 100,
                child: Icon(Icons.directions_run),
              ),
              childWhenDragging: Container(
                height: 100.0,
                width: 100.0,
                color: Colors.lightGreenAccent,
                child: Center(
                  child: Text("Child When Dragging"),
                ),
              ),
            ),
            DragTarget(
              builder: (
                BuildContext context,
                List<dynamic> accepted,
                List<dynamic> rejected,
              ) {
                return Container(
                  height: 100.0,
                  width: 100.0,
                  color: Colors.cyan,
                  child: Center(
                    child: Text("$acceptedData"),
                  ),
                );
              },
              onAccept: (int data) {
                setState(() {
                  acceptedData += data;
                });
              },
            )
          ],
        ),
      ),
    );
  }
}
```

**Comment by shihaohong** [comment]
I tested it and it works just fine when it's a StatefulWidget. Perhaps you made a mistake porting it into a stateful widget?

**Comment by parzuko** [comment]
> I tested it and it works just fine when it's a StatefulWidget. Perhaps you made a mistake porting it into a stateful widget?

Yes that might be it. I'll just push it with the `setState` then. Thanks!

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 9]
```diff
@@ -98,6 +98,70 @@ enum DragAnchor {
...
 /// {@youtube 560 315 https://www.youtube.com/watch?v=QzA4c4QHZCY}
+/// 
+/// {@tool dartpad --template=stateful_widget_material}
+///
+/// The following example has a [Draggable] widget along with a [DragTarget] 
+/// in a row demonstrating an incremented `acceptedData` integer value when 
+/// you drag the element to the target.
+///
+/// ```dart
+/// var acceptedData = 0;
```
Comment (shihaohong) [inline packages/flutter/lib/src/widgets/drag_target.dart:109]: Use types
```suggestion
/// int acceptedData = 0;
```

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 10]
```diff
@@ -98,6 +98,70 @@ enum DragAnchor {
...
+///         builder: (BuildContext context) {
+///             Container(
+///               height: 100.0,
+///               width: 100.0,
+///               color: Colors.cyan,
+///               child: Center(
+///                 child: Text("Value is updated to: $acceptedData"),
+///               ),
+///             ),
+///          }
```
Comment (shihaohong) [inline packages/flutter/lib/src/widgets/drag_target.dart:151]:
```suggestion
///          },
```

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 11]
```diff
@@ -98,6 +98,70 @@ enum DragAnchor {
...
+///          onAccept: (int data){
+///            setState((){
+///             acceptedData += data;
+///            });
+///          }
+///       )
+///     ]
```
Comment (shihaohong) [inline packages/flutter/lib/src/widgets/drag_target.dart:158]:
```suggestion
///          },
///       ),
///     ],
```

Comment (shihaohong) [reply packages/flutter/lib/src/widgets/drag_target.dart:158]: It also seems like the indentation here is a little off -- there's 3 spaces instead of 2 between this line and the next.

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 12]
```diff
@@ -98,6 +98,70 @@ enum DragAnchor {
...
+///       DragTarget(
+///         builder: (BuildContext context) {
+///             Container(
+///               height: 100.0,
+///               width: 100.0,
+///               color: Colors.cyan,
+///               child: Center(
+///                 child: Text("Value is updated to: $acceptedData"),
+///               ),
+///             ),
```
Comment (shihaohong) [inline packages/flutter/lib/src/widgets/drag_target.dart:150]: Indentation is 3 spaces instead of 2

**Review by shihaohong — COMMENTED** [review]
Some comments. Also, analyzer failing on trailing whitespace: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8864761594645604128/+/steps/prepare_environment/0/steps/Analyze/0/steps/analyze/0/stdout

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 13]
```diff
@@ -98,6 +98,70 @@ enum DragAnchor {
...
+///          }
+///          onAccept: (int data){
+///            setState((){
+///             acceptedData += data;
+///            });
+///          }
```
Comment (shihaohong) [inline packages/flutter/lib/src/widgets/drag_target.dart:156]:
```suggestion
///          onAccept: (int data) {
///            setState(() {
///             acceptedData += data;
///            });
///          }
```

**Comment by parzuko** [comment]
@shihaohong I've made the necessary changes. Apologies for the hassle.

**Comment by shihaohong** [comment]
Still a couple of trailing spaces:
```
▌08:23:37▐ Trailing spaces...
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
/tmp/flutter sdk/packages/flutter/lib/src/widgets/drag_target.dart:101: trailing U+0020 space character
/tmp/flutter sdk/packages/flutter/lib/src/widgets/drag_target.dart:104: trailing U+0020 space character
/tmp/flutter sdk/packages/flutter/lib/src/widgets/drag_target.dart:105: trailing U+0020 space character
```

One way to check before pushing the code is to run `flutter analyze --flutter-repo` to see if there are any outstanding analysis errors.

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 14]
```diff
@@ -98,6 +98,74 @@ enum DragAnchor {
...
+/// {@tool dartpad --template=stateful_widget_material}
+///
+/// The following example has a [Draggable] widget along with a [DragTarget] 
+/// in a row demonstrating an incremented `acceptedData` integer value when 
+/// you drag the element to the target.
+///
+/// ```dart
+///int acceptedData = 0;
+///
+///Widget build(BuildContext context) {
```
Comment (shihaohong) [inline packages/flutter/lib/src/widgets/drag_target.dart:111]: There should be at least one space between the comments and the sample code
```suggestion
/// int acceptedData = 0;
///
/// Widget build(BuildContext context) {
```

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 15]
```diff
@@ -98,6 +98,74 @@ enum DragAnchor {
...
+///  return Row(
+///      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
```
Comment (shihaohong) [inline packages/flutter/lib/src/widgets/drag_target.dart:113]: There is 4 spaces instead of 2 between these lines

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 16]
```diff
@@ -98,6 +98,74 @@ enum DragAnchor {
...
+///          },
+///          onAccept: (int data) {
+///            setState(() {
+///              acceptedData += data;
+///            });
+///          },
+///        ),
+///      ],
+///    );
+///}
```
Comment (shihaohong) [inline packages/flutter/lib/src/widgets/drag_target.dart:164]:
```suggestion
/// }
```

**Review by shihaohong — COMMENTED** [review]
The indentation is still off throughout the code sample. You can reference other code samples in the codebase if you're unsure about the indentation style.

Edit: It's at least one space from the start of the code comment `///`, and then an additional 2 spaces if it's a nested section of code.

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 17]
```diff
@@ -99,6 +99,74 @@ enum DragAnchor {
...
 ///
+/// {@tool dartpad --template=stateful_widget_material}
+///
+/// The following example has a [Draggable] widget along with a [DragTarget]
+/// in a row demonstrating an incremented `acceptedData` integer value when
+/// you drag the element to the target.
+///
+/// ```dart
+/// int acceptedData = 0;
+///
```
Comment (yash1200) [inline packages/flutter/lib/src/widgets/drag_target.dart:110]: Remove this extra space.

**Review by shihaohong — APPROVED** [review]
LGTM. Thanks for the contribution!

**Comment by johnpryan** [comment]
@parzuko Thanks for participating in AdoptAWidget. We would like to send you a special thanks. If you are interested, please send an email to nyener at google dot com and provide a link to this issue.
